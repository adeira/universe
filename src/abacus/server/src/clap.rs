use clap::{Arg, ArgSettings, Command, ValueHint};

pub fn generate_clap_app() -> Command<'static> {
    clap::command!()
        .arg(
            Arg::new("no-migrations")
                .long("no-migrations")
                .takes_value(false)
                .help("Skips database migrations"),
        )
        .arg(
            Arg::new("arangodb-url")
                .long("arangodb-url")
                .takes_value(true)
                .default_value("http://127.0.0.1:8529/")
                .value_hint(ValueHint::Url),
        )
        .arg(
            Arg::new("arangodb-database")
                .long("arangodb-database")
                .takes_value(true)
                .default_value("abacus"),
        )
        .arg(
            Arg::new("arangodb-username")
                .long("arangodb-username")
                .takes_value(true)
                .default_value("abacus"),
        )
        .arg(
            Arg::new("arangodb-password")
                .long("arangodb-password")
                .takes_value(true)
                .default_value(""),
        )
        .arg(
            Arg::new("stripe-restricted-api-key")
                .long("stripe-restricted-api-key")
                .help("Restricted Stripe.com API key (prefixed by 'rk_*')")
                .long_help(
                    "Restricted Stripe.com API key (prefixed by 'rk_*') to be used when calling \
                    Stripe.com APIs. Secret API key should never be used directly in this application. \
                    More information: https://stripe.com/docs/keys#limit-access",
                )
                .takes_value(true),
        )
        .arg(
            Arg::new("stripe-webhook-secret")
                .long("stripe-webhook-secret")
                .help("Secret key for webhooks verification.")
                .long_help(
                    "Stripe generates a unique secret key for each webhooks endpoint. It is being \
                    used for verifying the webhook payload signature to make sure that only \
                    Stripe.com can send these payloads.",
                )
                .takes_value(true),
        ).subcommand(
            Command::new("generate-cli-completions")
                .override_help("Generate CLI completions for specified shells.")
                .arg(
                    Arg::new("shell")
                        .long("shell")
                        .takes_value(true)
                        .possible_values(&["bash", "zsh"])
                        .setting(
                            ArgSettings::Required
                        )
                )
        )
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn generated_clap_usage_test_long_help() {
        let mut output = Vec::new();
        generate_clap_app().write_long_help(&mut output).unwrap();

        insta::with_settings!({snapshot_path => "__snapshots__"}, {
            insta::assert_snapshot!(String::from_utf8(output).unwrap())
        });
    }

    #[test]
    fn generated_clap_usage_test_short_help() {
        let mut output = Vec::new();
        generate_clap_app().write_help(&mut output).unwrap();

        insta::with_settings!({snapshot_path => "__snapshots__"}, {
            insta::assert_snapshot!(String::from_utf8(output).unwrap())
        });
    }
}
