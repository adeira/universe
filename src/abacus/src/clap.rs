use clap::{App, ArgSettings};

pub fn generate_clap_app() -> App<'static> {
    clap::app_from_crate!()
        .arg(
            clap::Arg::new("no-migrations")
                .long("no-migrations")
                .takes_value(false)
                .about("Skips database migrations"),
        )
        .arg(
            clap::Arg::new("arangodb-url")
                .long("arangodb-url")
                .takes_value(true)
                .default_value("http://127.0.0.1:8529/"),
        )
        .arg(
            clap::Arg::new("arangodb-database")
                .long("arangodb-database")
                .takes_value(true)
                .default_value("abacus"),
        )
        .arg(
            clap::Arg::new("arangodb-username")
                .long("arangodb-username")
                .takes_value(true)
                .default_value("abacus"),
        )
        .arg(
            clap::Arg::new("arangodb-password")
                .long("arangodb-password")
                .takes_value(true)
                .settings(&[
                    ArgSettings::AllowEmptyValues, // Allows an arg accept empty values such as `""`
                ])
                .default_value(""),
        )
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn generated_clap_usage_test() {
        let mut output = Vec::new();
        generate_clap_app().write_long_help(&mut output).unwrap();

        insta::with_settings!({snapshot_path => "__snapshots__"}, {
            insta::assert_snapshot!(String::from_utf8(output).unwrap())
        });
    }
}
