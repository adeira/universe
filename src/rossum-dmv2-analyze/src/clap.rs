use clap::{Arg, Command};

pub fn generate_clap_app() -> Command {
    clap::command!()
        .arg(
            Arg::new("concurrency")
                .short('c')
                .long("concurrency")
                .num_args(1)
                .long_help("How many annotations should be analyzed concurrently.")
                .required(false)
                .default_value("100"),
        )
        .arg(
            Arg::new("config-file")
                .long("config-file")
                .num_args(1)
                .long_help(
                    "JSON config file with the new DMv2 configuration. Copy-paste here the whole \
                    content of DMv2 configuration including the top-level `configurations` key.",
                )
                .required(true),
        )
        .arg(
            Arg::new("dm-hook-id")
                .long("dm-hook-id")
                .num_args(1)
                .long_help("DMv2 hook ID to analyze against.")
                .required(true),
        )
        .arg(
            Arg::new("queue-id")
                .long("queue-id")
                .num_args(1)
                .long_help(
                    "Queue ID that will be used for the analysis of all relevant annotations.",
                )
                .required(true),
        )
        .arg(
            Arg::new("api-token")
                .long("api-token")
                .num_args(1)
                .long_help(
                    "API token belonging to the organization that will be used for calling Rossum API endpoints.",
                )
                .required(true),
        )
}
