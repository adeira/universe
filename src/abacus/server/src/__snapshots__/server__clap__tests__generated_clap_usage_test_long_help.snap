---
source: server/src/clap.rs
assertion_line: 79
expression: generate_clap_app().render_long_help().to_string()
---
Usage: server [OPTIONS] [COMMAND]

Commands:
  generate-cli-completions  
  help                      Print this message or the help of the given subcommand(s)

Options:
      --no-migrations
          Skips database migrations

      --arangodb-url <arangodb-url>
          [default: http://127.0.0.1:8529/]

      --arangodb-database <arangodb-database>
          [default: abacus]

      --arangodb-username <arangodb-username>
          [default: abacus]

      --arangodb-password <arangodb-password>
          [default: ]

      --stripe-restricted-api-key <stripe-restricted-api-key>
          Restricted Stripe.com API key (prefixed by 'rk_*') to be used when calling Stripe.com APIs. Secret API key should never be used directly in this application. More information: https://stripe.com/docs/keys#limit-access
          
          [env: STRIPE_RESTRICTED_API_KEY=]

      --stripe-webhook-secret <stripe-webhook-secret>
          Stripe generates a unique secret key for each webhooks endpoint. It is being used for verifying the webhook payload signature to make sure that only Stripe.com can send these payloads.
          
          [env: STRIPE_WEBHOOK_SECRET=]

  -h, --help
          Print help (see a summary with '-h')

  -V, --version
          Print version

